dotcom-rendering/src/web/lib/liveblogAdSlots.ts

```diff
@@ -28,49 +38,57 @@ type BlockElementHeight = {
 
 /**
  * Approximations of the size each block element type will take up on the page in pixels.
- * Predictions are made for desktop, as we prefer to use a value closer to the lower bound,
- * as we would rather display too few ads than too many. Practically, this will mean that we
- * will show the the right frequency of ads on desktop and too few on smaller devices.
+ * Predictions are made for mobile viewports, as data suggests that the majority of liveblog page
+ * views are made using mobile devices.
  */
+const blockquoteElementHeightData: BlockElementHeight = {
+	type: 'model.dotcomrendering.pageElements.BlockquoteBlockElement',
+	elementHeightExcludingText: 0,
+	textHeight: {
+		lineHeight: 25.5,
+		lineLength: 40,
+	},
+};
+
 const commentElementHeightData: BlockElementHeight = {
 	type: 'model.dotcomrendering.pageElements.CommentBlockElement',
-	elementHeightExcludingText: 64,
+	elementHeightExcludingText: 74,
 	textHeight: {
-		lineHeight: 23,
-		lineLength: 70,
+		lineHeight: 18,
+		lineLength: 28,
```

Comment:
I'm finding these numbers a bit hard to parse as separate consts. Would it be possible and fit with the current flow to have a centralised map where they can be looked up? This could then sit in its own file and be exported.

Something like:
```js
{
  'model.dotcomrendering.pageElements.CommentBlockElement' : { 
    elementHeightExcludingText:  74,
    textHeight: {
      lineHeight: 18,
      lineLength: 28
    }
  }
  'model.dotcomrendering.pageElements.EmbedBlockElement' : {
    ...
  }
}
```

Reply:
Yeah agreed. I have made this change.

Reply:
Now that this is in place I wonder if there is potential for more simplification.

Could we:

1. use the `CAPIElement._type` property as the key into this map
2. add an optional function to the map to retrieve the appropriate text property for that element
3. `calculateElementHeight` can check if the element has a text function and if so use that

```js
{
  'model.dotcomrendering.pageElements.TextBlockElement' : { 
    heightExcludingText: 0,
    textHeight: {
      lineHeight: 25.5,
      lineLength: 39,
    },
    text: (element) => element.html.replace(/<[^>]+>/g, '')
  },  
  ...
}
```

Which _I think_ would mean `calculateElementSize` doesn't need the switch-case for every type and can call:

```js
calculateElementHeight(elementHeightData[element._type])
```

`TableBlockElement` and the unknown type would need to be handled separately but I think they are the only exceptional cases.